sdk/core/Azure.Core/src/Pipeline/Policies/RetryPolicy.cs

```diff
@@ -10,16 +10,49 @@
 
 namespace Azure.Core.Pipeline.Policies
 {
-    public abstract class RetryPolicy : HttpPipelinePolicy
+    public class RetryOptions
```

Comment:
Personally, I'd be interested in seeing a shared construct that provides the full framework for retry policies,  from the options to calculating the next delay.   

A consistent approach to calculations that could be overridden would help to normalize across HTTP and non-HTTP clients.

Reply:
It can be done based on these types but without adding public APIs. We can have a shared-code type that takes in one of the option types and provides reusable delay calculation routines.